File: week18/김고은/수리공 항승/1449.py
import sys

input = sys.stdin.readline

N,L = map(int, input().split())
inputs = list(map(int, input().split()))

input = inputs.sort()
coverd_areea = 0
ans = 0

for i in range (0,N):
    if(coverd_areea > inputs[i]):
        continue
    ans+=1
    coverd_areea = inputs[i] + L         
    

print(ans)

File: week18/김고은/수리공 항승/1449.js
const fs = require('fs')
const filePath = process.platform === 'linux'? '/dev/stdin' : __dirname + '/input.txt'
const input = fs.readFileSync(filePath).toString().trim().split('\n');

let [N, L] = input[0].split(' ').map(Number);
let points = input[1].split(' ').map(Number);

let ans = 0;
let taped_count = 0;

points.sort((a,b) => a-b);

for(let i =0; i<N; i++){
    if(taped_count > points[i]) continue;

    ans++;
    taped_count = points[i] + L;
}

console.log(ans)